packages/library-base/src/commonMain/kotlin/io/realm/internal/dynamic/DynamicMutableRealmImpl.kt

```diff
@@ -79,10 +82,10 @@ internal open class DynamicMutableRealmImpl(
             null
         } else if (!obj.isManaged()) {
```

Comment:
If we rearranged this to use `obj.asObjectReference()?. { ... } ?: throw ` then we have immediate access to all the internals in the block and don't need to mess with `RealmObjectInternal` too!?